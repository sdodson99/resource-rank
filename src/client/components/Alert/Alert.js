import React from 'react';
import PropTypes from 'prop-types';
import styles from './Alert.module.css';

const Alert = ({ children, icon, border, className }) => {
  const calculateClassName = () => {
    let totalClassName = styles.Alert;

    if (border) {
      totalClassName += ' border rounded';
    }

    if (className) {
      totalClassName += ` ${className}`;
    }

    return totalClassName;
  };

  return (
    <div className={calculateClassName()} data-testid="Alert">
      {icon && <div className="mr-2 inline-block align-middle">{icon}</div>}
      <div className="inline-block align-middle">{children}</div>
    </div>
  );
};

Alert.propTypes = {
  children: PropTypes.node,
  icon: PropTypes.node,
  border: PropTypes.bool,
  className: PropTypes.string,
};

Alert.defaultProps = {};

export default Alert;
