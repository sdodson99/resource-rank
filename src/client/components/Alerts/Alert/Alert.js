import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import styles from './Alert.module.css';
import { useIntersectionObserver } from 'react-intersection-observer-hook';

const Alert = ({ children, icon, border, className, scrollTo }) => {
  const [alertRef, { entry }] = useIntersectionObserver();
  const isVisible = entry?.isIntersecting;

  const [hasScrolledIntoView, setHasScrolledIntoView] = useState(false);
  useEffect(() => {
    setHasScrolledIntoView(false);
  }, [scrollTo]);
  useEffect(() => {
    if (!scrollTo) {
      return;
    }

    if (hasScrolledIntoView) {
      return;
    }

    if (!entry) {
      return;
    }

    if (!isVisible) {
      entry.target.scrollIntoView();
    }

    setHasScrolledIntoView(true);
  }, [scrollTo, isVisible, hasScrolledIntoView, entry]);

  const calculateClassName = () => {
    let totalClassName = styles.Alert;

    if (border) {
      totalClassName += ' border rounded';
    }

    if (className) {
      totalClassName += ` ${className}`;
    }

    return totalClassName;
  };

  return (
    <div ref={alertRef} className={calculateClassName()} data-testid="Alert">
      {icon && <div className="mr-2 inline-block align-middle">{icon}</div>}
      <div className="inline-block align-middle">{children}</div>
    </div>
  );
};

Alert.propTypes = {
  children: PropTypes.node,
  icon: PropTypes.node,
  border: PropTypes.bool,
  className: PropTypes.string,
  scrollTo: PropTypes.bool,
};

Alert.defaultProps = {};

export default Alert;
