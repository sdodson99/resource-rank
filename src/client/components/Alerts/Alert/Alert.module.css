.Alert {
  @apply p-4;
}
