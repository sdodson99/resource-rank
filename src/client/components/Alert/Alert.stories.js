import React from 'react';
import Alert from './Alert';
import { Info } from '@material-ui/icons';

export default {
  component: Alert,
  title: 'Components/Alert',
};

const Template = (args) => <Alert {...args}>This is an alert!</Alert>;

export const Default = Template.bind({});
Default.args = {};

export const Icon = Template.bind({});
Icon.args = {
  icon: <Info htmlColor="red" />,
};

export const Custom = Template.bind({});
Custom.args = {
  border: true,
  className: 'bg-blue-50 border-blue-100 text-blue-800',
  icon: <Info color="primary" />,
};
Custom.displayName = 'Custom';
