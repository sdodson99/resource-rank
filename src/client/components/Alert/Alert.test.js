import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Alert from './Alert';
import { createRenderer } from 'react-test-renderer/shallow';

describe('<Alert />', () => {
  it('should mount', () => {
    render(<Alert />);

    const alert = screen.getByTestId('Alert');

    expect(alert).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const page = createRenderer().render(<Alert>Message</Alert>);

    expect(page).toMatchSnapshot();
  });

  describe('with border', () => {
    it('should render correctly', () => {
      const page = createRenderer().render(
        <Alert border={true}>Message</Alert>
      );

      expect(page).toMatchSnapshot();
    });
  });

  describe('with className', () => {
    it('should render correctly', () => {
      const page = createRenderer().render(
        <Alert className={'border-red-700'}>Message</Alert>
      );

      expect(page).toMatchSnapshot();
    });
  });

  describe('with icon', () => {
    it('should render correctly', () => {
      const page = createRenderer().render(
        <Alert icon={<div>Icon</div>}>Message</Alert>
      );

      expect(page).toMatchSnapshot();
    });
  });
});
