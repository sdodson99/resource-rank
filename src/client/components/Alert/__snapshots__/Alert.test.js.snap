// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should render correctly 1`] = `
<div
  className="Alert"
  data-testid="Alert"
>
  <div
    className="inline-block align-middle"
  >
    Message
  </div>
</div>
`;

exports[`<Alert /> with border should render correctly 1`] = `
<div
  className="Alert border rounded-sm"
  data-testid="Alert"
>
  <div
    className="inline-block align-middle"
  >
    Message
  </div>
</div>
`;

exports[`<Alert /> with className should render correctly 1`] = `
<div
  className="Alert border-red-700"
  data-testid="Alert"
>
  <div
    className="inline-block align-middle"
  >
    Message
  </div>
</div>
`;

exports[`<Alert /> with icon should render correctly 1`] = `
<div
  className="Alert"
  data-testid="Alert"
>
  <div
    className="mr-2 inline-block align-middle"
  >
    <div>
      Icon
    </div>
  </div>
  <div
    className="inline-block align-middle"
  >
    Message
  </div>
</div>
`;
